pkg/gatherers/conditional/filter_and_gather_container_logs.go

```diff
@@ -84,75 +85,72 @@ func gatherContainerLogs(
 
 	var sendWG sync.WaitGroup
 	for _, logRequest := range namespaceToLogRequestMap {
-		klog.Infof("Start checking namespace %s for the Pod name pattern %s\n", logRequest.Namespace, logRequest.PodNameRegex)
-
-		for podNameRegex := range logRequest.PodNameRegex {
-			sendWG.Add(1)
-			go filterContainerLogs(shorterCtx, coreClient, logRequest, podNameRegex, &sendWG, recCh)
-		}
+		sendWG.Add(1)
+		klog.Infof("Start checking namespace %s for the Pod name pattern %s\n",
+			logRequest.Namespace,
+			mapKeysToSlice(logRequest.PodNameRegexToMessages))
+		go filterContainerLogs(shorterCtx, coreClient, logRequest, &sendWG, recCh)
 	}
 	sendWG.Wait()
 	close(recCh)
 	receiveWG.Wait()
 	return records, errs
 }
 
+func mapKeysToSlice(m map[PodNameRegexPrevious]sets.Set[string]) []string {
+	var keys []string
```

Comment:
knowing the map length wouldn't be better (efficient) to use `make([]string, len(m))` and then assign values instead of appending?

Reply:
:+1: yes. Thank you.